posthog/migrations/0204_global_tags_setup.py

```diff
@@ -0,0 +1,205 @@
+# Generated by Django 3.2.5 on 2022-02-07 21:37
+
+import django.contrib.postgres.fields
+import django.db.models.deletion
+from django.db import migrations, models
+
+import posthog.models.utils
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0203_dashboard_permissions"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="Tag",
+            fields=[
+                (
+                    "id",
+                    models.UUIDField(
+                        default=posthog.models.utils.UUIDT, editable=False, primary_key=True, serialize=False
+                    ),
+                ),
+                ("name", models.SlugField()),
+                ("team", models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to="posthog.team")),
+            ],
+        ),
+        migrations.RenameField(model_name="dashboard", old_name="tags", new_name="deprecated_tags",),
+        migrations.RenameField(model_name="insight", old_name="tags", new_name="deprecated_tags",),
+        migrations.AlterField(
+            model_name="dashboard",
+            name="deprecated_tags",
+            field=django.contrib.postgres.fields.ArrayField(
+                base_field=models.CharField(max_length=32),
+                blank=True,
+                db_column="tags",
```

Comment:
We're doing this for backwards compatibility as an alternative to renaming the column entirely which makes things unnecessarily complicated. i.e. when a service fetches tags using an outdated SQL query that has no knowledge of this new tagging model. That being said, I don't think this is going to be too much of a problem since PH is rolled out as a monolith, but users who _might_ have custom scripts might be affected. Also there's no technical necessity to rename the column name 🤷